packages/theme/src/unified/UnifiedThemeProvider.tsx

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { ReactNode } from 'react';
+import './MuiClassNameSetup';
+import { ThemeProvider } from '@material-ui/core/styles';
+import {
+  StyledEngineProvider,
+  ThemeProvider as Mui5Provider,
+} from '@mui/material/styles';
+import Mui5CssBaseline from '@mui/material/CssBaseline';
+import { UnifiedTheme } from './types';
+import { CssBaseline } from '@material-ui/core';
+
+/**
+ * Props for {@link UnifiedThemeProvider}.
+ *
+ * @public
+ */
+export interface UnifiedThemeProviderProps {
+  children: ReactNode;
+  theme: UnifiedTheme;
+  noCssBaseline?: boolean;
+}
+
+/**
+ * Provides themes for all MUI versions supported by the provided unified theme.
+ *
+ * @public
+ */
+export function UnifiedThemeProvider(
+  props: UnifiedThemeProviderProps,
+): JSX.Element {
+  const { children, theme, noCssBaseline = false } = props;
+
+  const v4Theme = theme.getTheme('v4');
+  const v5Theme = theme.getTheme('v5');
+
+  let cssBaseline: JSX.Element | undefined = undefined;
+  if (!noCssBaseline) {
+    if (v5Theme) {
+      cssBaseline = <Mui5CssBaseline enableColorScheme />;
+    } else if (v4Theme) {
```

Comment:
Is this reachable? Won't all themes using the `UnifiedThemeProvider` provide a v5 theme by default?

1. If the `UnifiedThemeProvider` is used, the `theme` passed in as a prop must be a `UnifiedTheme` itself
2. Therefor it must have been generated by `createUnifiedTheme` or `createUnifiedThemeFromV4`
3. This means there will always be at least the v5 theme
      
Additionallly looking at the code of v5 `CSSBaseline` (https://github.com/mui/material-ui/blob/master/packages/mui-material/src/CssBaseline/CssBaseline.js) it seems to me that the main difference between v4 & v5 is that you can optionally enable the css `color-scheme` property and visually the font size (https://mui.com/material-ui/migration/v5-component-changes/#update-body-font-size), which we override in `v5/defaultComponentTheme.ts` was changed.